tests/framework/microvm.py

```diff
@@ -1007,13 +1013,16 @@ def ssh_iface(self, iface_idx=0):
         """Return a cached SSH connection on a given interface id."""
         guest_ip = list(self.iface.values())[iface_idx]["iface"].guest_ip
         self.ssh_key = Path(self.ssh_key)
-        return net_tools.SSHConnection(
+        connection = net_tools.SSHConnection(
             netns=self.netns.id,
             ssh_key=self.ssh_key,
             user="root",
             host=guest_ip,
+            control_path=Path(self.chroot()) / f"ssh-{iface_idx}.sock",
```

Comment:
Do I understand correctly that now instead of openning new host<->guest ssh connections, we open 1 master ssh on first `ssh_iface` call and then create new ssh processes that just communicate with guest through the master ssh?

Reply:
yep!

Reply:
simplification...